### ISSUE TITLE
#[deny(warnings)] does not change the lint source, produces confusing error

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#30730]
Normally, an unused import:

```rust
use std::thread;
fn main() { }
```

Produces this compiler output:

```
linttest.rs:4:5: 4:16 warning: unused import, #[warn(unused_imports)] on by default
linttest.rs:4 use std::thread;
```

Saying the 'unused_imports' lint is 'on by default'.

If you explicitly set *that lint* to deny:

```rust
#![deny(unused_imports)]
use std::thread;
fn main() { }
```

Then the error indicates where the lint was set to 'deny':

```
linttest.rs:4:5: 4:16 error: unused import
linttest.rs:4 use std::thread;
                  ^~~~~~~~~~~
linttest.rs:1:9: 1:23 note: lint level defined here
linttest.rs:1 #![deny(unused_imports)]
                      ^~~~~~~~~~~~~~
error: aborting due to previous error
```

If you instead change warnings to errors:

```rust
#![deny(warnings)]
use std::thread;
fn main() { }
```

Then you get this error:

```
linttest.rs:4:5: 4:16 error: unused import, #[deny(unused_imports)] on by default
linttest.rs:4 use std::thread;
                  ^~~~~~~~~~~
error: aborting due to previous error
```

Which says 'deny(unused_imports)' is 'on by default'. That is not true. unused_imports is warn by default. This case should point to `#[deny(warnings)]` as the source of the error.

Discovered while interpreting all the lint regressions in [a recent crater report](https://internals.rust-lang.org/t/regression-report-stable-2015-12-10-vs-nightly-2015-12-21/3026).

cc @nikomatsakis

**Comment by shssoichiro** [comment]
I took a shot at this and made a PR. Now your example code for deny(warnings) shows the following:

    test_deny.rs:2:5: 2:16 error: unused import
    test_deny.rs:2 use std::thread;
                       ^~~~~~~~~~~
    test_deny.rs:1:9: 1:17 note: lint level defined here
    test_deny.rs:1 #![deny(warnings)]
                           ^~~~~~~~
    error: aborting due to previous error

**Comment by sanxiyn** [comment]
This was fixed by #30804.
